interbotix_ros_slate/interbotix_slate_driver/src/slate_base.cpp

```diff
@@ -76,135 +72,96 @@ SlateBase::SlateBase(const rclcpp::NodeOptions & options)
     "set_light_state",
     std::bind(&SlateBase::set_light_state_callback, this, _1, _2, _3));
 
-  std::string dev;
-  if (!base_driver::chassisInit(dev)) {
-    RCLCPP_ERROR(get_logger(), "Failed to initialize base port.");
-    ::exit(EXIT_FAILURE);
-  }
-  RCLCPP_INFO(get_logger(), "Initalized base at port '%s'.", dev.c_str());
-  char version[32] = {0};
-  if (base_driver::getVersion(version)) {
-    RCLCPP_INFO(get_logger(), "Base version: %s", version);
-  }
+  std::string result;
+  init_base(result);
+  RCLCPP_ERROR(get_logger(), result.c_str());
 }
 
 void SlateBase::update()
 {
   rclcpp::spin_some(get_node_base_interface());
   current_time_ = get_clock()->now();
 
-  // time out velocity commands
-  if (current_time_ - cmd_vel_time_last_update_ > cmd_vel_timeout_) {
-    cmd_vel_x_ = 0.0f;
-    cmd_vel_z_ = 0.0f;
-  }
-
-  // limit velocity commands
-  cmd_vel_x_ = std::min(max_vel_x_, std::max(-max_vel_x_, cmd_vel_x_));
-  cmd_vel_z_ = std::min(max_vel_z_, std::max(-max_vel_z_, cmd_vel_z_));
-
-  // initialize chassis data and use it to update the driver
-  base_driver::ChassisData data = {
-    .cmd_vel_x=cmd_vel_x_,
-    .cmd_vel_y=0.0,
-    .cmd_vel_z=cmd_vel_z_,
-    .light_state=light_state_,
-    .system_state=0};
-
-  if (!base_driver::updateChassisInfo(&data)) {
+  if (!base_driver::updateChassisInfo(&data_)) {
     return;
   }
 
-  // extract and update base system command bytes
-  sys_cmd_ = data.cmd;
-
-  // update battery state every 10 iterations
+  // Update battery state every 10 iterations
   cnt_++;
   auto battery_state = BatteryState();
   if (cnt_ % 10 == 0) {
     battery_state.header.stamp = current_time_;
-    battery_state.voltage = data.voltage;
-    battery_state.current = data.current;
-    battery_state.percentage = data.charge;
-    battery_state.power_supply_status = data.system_state;
+    battery_state.voltage = data_.voltage;
+    battery_state.current = data_.current;
+    battery_state.percentage = data_.charge;
+    battery_state.power_supply_status = data_.system_state;
```

Comment:
Remove this system_state line and set all unmeasured fields to NaN as per [the battery state message definition](https://github.com/ros2/common_interfaces/blob/rolling/sensor_msgs/msg/BatteryState.msg#L38-L43).